src/main/java/com/genesys/workspace/Notifications.java

```diff
@@ -103,6 +118,11 @@ public void disconnect() throws WorkspaceApiException  {
     }
     
     public void subscribe(String channelName, NotificationListener listener) {
-        listeners.computeIfAbsent(channelName, k -> new ConcurrentLinkedQueue<>()).add(listener);
+        if(listeners.get(channelName) == null){
+            Collection<NotificationListener> notificationListeners = new ConcurrentLinkedQueue<>();
+            notificationListeners.add(listener);
+            this.listeners.put(channelName,notificationListeners);
+
+        }
```

Comment:
broken logic, should be: 
list = listeners.get(channelName)
if(list == null) {
   list = create new 
   listeners.put(list)
}
list.add(listener)